packages/documentation-v7/src/stories/components/select/select.stories.ts

```diff
@@ -0,0 +1,328 @@
+        step: 1,
+      },
+      table: {
+        category: 'General',
+      },
+    },
+    hint: {
+      name: 'Helper Text',
+      description: 'Text to place in the help text area of the component.',
+      control: {
+        type: 'text',
+      },
+      table: {
+        category: 'General',
+      },
+    },
+    disabled: {
+      name: 'Disabled',
+      description:
+        'When set to `true`, disables the component\'s functionality and places it in a disabled state.<span className="mt-mini alert alert-info alert-sm">There are accessibility issues with the disabled state.<br/>Please read our <a href="/?path=/docs/foundations-accessibility--page#disabled-state">disabled state accessibility guide</a>.</span>',
+      control: {
+        type: 'boolean',
+      },
+      table: {
+        category: 'States',
+      },
+    },
+    validation: {
+      name: 'Validation',
+      description: 'Controls the validation state appearance of the component.',
+      control: {
+        type: 'radio',
+        labels: {
+          'null': 'Default',
+          'is-valid': 'Valid',
+          'is-invalid': 'Invalid',
+        },
+      },
+      options: ['null', 'is-valid', 'is-invalid'],
+      table: {
+        category: 'States',
+      },
+    },
+  },
+};
+
+export default meta;
+
+type Story = StoryObj;
+
+const Template: Story = {
+  render: (args: Args, context: StoryContext) => {
+    const [_, updateArgs] = useArgs();
+    const id = `${context.viewMode}_${context.story.replace(/\s/g, '-')}_ExampleSelect`;
+    const classes = ['form-select', args.size, args.validation]
+      .filter(c => c && c !== 'null')
+      .join(' ');
+    const useAriaLabel = !args.floatingLabel && args.hiddenLabel;
+    const label = !useAriaLabel
+      ? html`
+          <label for="${id}" class="form-label">${args.label}</label>
+        `
+      : null;
+    const optionElements = Array.from({ length: args.options - 1 }, (_, i) => i + 2).map(
+      (key: number) => html`
+        <option value="valoro_${key}">Opcion ${key}</option>
+      `,
+    );
+    const options = [
+      html`
+        <option>Elektu opcion...</option>
+      `,
+      ...optionElements,
+    ];
+    const contextuals = [
+      args.validation === 'is-valid'
+        ? html`
+            <p class="valid-feedback">Ggranda sukceso!</p>
+          `
+        : null,
+      args.validation === 'is-invalid'
+        ? html`
+            <p class="invalid-feedback">Eraro okazis!</p>
+          `
+        : null,
+      args.hint !== ''
+        ? html`
+            <div class="form-text">${args.hint}</div>
+          `
+        : null,
+    ];
+    function onChange(e: Event) {
+      updateArgs({ value: (e.target as HTMLSelectElement).value });
+
+      if (document.activeElement === e.target) {
+        setTimeout(() => {
+          const element: HTMLSelectElement | null = document.querySelector(`#${id}`);
+          if (element) element.focus();
+        }, 25);
+      }
```

Comment:
These lines are setting the focus programmatically to the select field. It's probably not needed anymore in Storybook version 7.
Remove them and check if the focus stays on the select after changing its value in the story. If this is the case, we can get rid of these lines and inline the onChange function again into the control as a one-liner:

```ts
@change="${(e: Event) => updateArgs({ value: (e.target as HTMLSelectElement).value })}"
```

Reply:
You are correct it works without this part of the function.